api/proto/teleport/legacy/types/types.proto

```diff
@@ -6451,8 +6451,15 @@ message PluginEntraIDSyncSettings {
   // DefaultOwners are the default owners for all imported access lists.
   repeated string default_owners = 1;
 
-  // SSOConnectorID is the name of the Teleport SSO connector created and used by the Entra ID plugin
+  // SSOConnectorID is the name of the Teleport SSO connector created and used by the Entra ID plugin.
   string sso_connector_id = 2;
+
+  // use_system_credentials uses the system available credentials instead of using the OIDC flow to authenticate
+  // to Azure EntraId. This is important for clusters that are not available through internet.
+  bool use_system_credentials = 3;
+
+  // tenant_id is the Azure Directory this plugin syncs.
```

Comment:
Is this only used when use_system_credentials=true? Please document when we expect/require it to be set.

Reply:
No. This value was stored in another object and from now on will also be stored in the plugin object. 
Because of compatibility reasons, I can't enforce it in the check and set defaults otherwise existing plugins will fail but I have a backfill logic in `e`

Reply:
Can we say anything else about when it's expected to be set?

Reply:
Done